pkg/repository/profile/repository.go

```diff
@@ -0,0 +1,46 @@
+package profile_repository
+
+import (
+	"OnboardingExercise/pkg/custom_errors"
+	"fmt"
+)
+
+// Repository Note: I thought about another solution for the address changed case
+// in a regular case the Repository won't contain something that should change
+// it will contain a constant db client
+// so maybe a better solution will be to extract the profiles slice from the repository and make it global?
+type Repository struct {
+	profiles []Profile // will be changed to db client
+}
+
+func NewDAL(profiles []Profile) Repository {
+	return Repository{profiles: profiles}
+}
+
+func (repository *Repository) GetAllProfiles() []Profile {
+	return repository.profiles
+}
+
+func (repository *Repository) GetProfileByUserID(userId string) (Profile, error) {
+	for _, profile := range repository.GetAllProfiles() {
+		if profile.UserId == userId {
+			return profile, nil
+		}
+	}
+
+	return Profile{}, custom_errors.NotFoundError{Err: fmt.Sprintf("user %s does not exist", userId)}
```

Comment:
I'd use errors.Wrap so we will have a stack trace

Reply:
I added the stack trace to the error initlaization